### DISCUSSION TITLE
no internet connection message

### DISCUSSION THREAD

**Discussion by b1k3rdude** [ungoogled-software/ungoogled-chromium-android#106] [General]
So added the repo to F-droid, installed UC and upon running the app I am seeing a message at the top of the browser window "no internet connection" when it does in fact have an internet connection.

**Comment by wchen342** [comment]
Chromium get internet connection information from Android system directly. If somehow the system's probing of internet connection is blocked then it will show up in chromium as no internet connection too.

**Comment by b1k3rdude** [comment]
Samsung browser doesn't show this error, and I will have to check but I don't think chrome does either.

**Reply by wchen342** [reply]
I see. It's hard to tell what else can cause this. I saw someone talked about VPN causing this but I assume you are not using one? Another thing to check is whether this happens on Bromite too since we share some of the patches.

**Reply by b1k3rdude** [reply]
Checking the Bromite build, same issue. I am seeing it on wifi and data.

I, like I imagine a lot of users dont need a browser (or the OS for that matter) checking to see if there is an internet connection, I am perfectly capable of doing that myself.  How would I put a request in to get this functionality removed?

**Reply by wchen342** [reply]
Can you also test on a plain chromium to see whether it has this problem or not? You can get one [here](https://www.googleapis.com/download/storage/v1/b/chromium-browser-snapshots/o/Android%2F920003%2Fchrome-android.zip?alt=media).

We did not add this functionality, it comes with chromium base and the network connection probing is in Android itself. What we can do here is probably adding a flag to remove the call to this system functionality, but it won't stop other apps on your Android phone and Android itself from sending the network probe. For that you will need a custom ROM that removes corresponding codes from the system itself.

On a more technical note, @uazo @csagan5 I know you have [dns-send-IPv6-connectivity-probes-to-RIPE-DNS.patch](https://github.com/bromite/bromite/blob/master/build/patches/dns-send-IPv6-connectivity-probes-to-RIPE-DNS.patch) to switch DNS probe but it seems that patch is unrelated in this case since I don't use that patch. Have you seen similar bug reports?

**Reply by b1k3rdude** [reply]
Thats fair enough, I only have one other app that shows a related problem and I already have a work around for that. So what ever you can do would be appreciated :-)

And yeah, I wish I could disable the network probe thing in the OS entirely. Its completely unnecessary and just another thing that google is doing behind the scenes to increase non-consented telemetry.

**Reply by csagan5** [reply]
> On a more technical note, @uazo @csagan5 I know you have [dns-send-IPv6-connectivity-probes-to-RIPE-DNS.patch](https://github.com/bromite/bromite/blob/master/build/patches/dns-send-IPv6-connectivity-probes-to-RIPE-DNS.patch) to switch DNS probe but it seems that patch is unrelated in this case since I don't use that patch. Have you seen similar bug reports?

Yes that patch is unrelated; there might have been similar bug reports but I think the functionality being used here by Chromium (and forks) is related to captive network portal check, it is the same that allows to login when the wifi access point requires a login before granting internet access. It is not connecting to Google servers IIRC but uses a standard probe on the network.

**Reply by uazo** [reply]
> Chromium get internet connection information from Android system directly.

the same thing happened to me because of the firewall (I use AFWall+), it was enough for me to re-enable internet access to an Android service (sorry but I don't remember which one) to fix it. it's probably the same thing happening to you.

**Reply by b1k3rdude** [reply]
> 
> 
> > Chromium get internet connection information from Android system directly.
> 
> the same thing happened to me because of the firewall (I use AFWall+), it was enough for me to re-enable internet access to an Android service (sorry but I don't remember which one) to fix it. it's probably the same thing happening to you.

I also use AFWall+, but I run whitelist mode. Its probably 'Android system ID:1000', and that has way, way too many built-in/connected services to be allowed to access the internet. I have no idea how to just give individual services access through the firewall, or what telemetry said service/s spit out.

**Reply by uazo** [reply]
> Its probably 'Android system ID:1000'

yes, I know, it depends on the roms, you have to act in the AndroidManifest.xml to split the UID (but it's complex and I haven't tried it either).
I suggest you to try with https://trackercontrol.org/

**Reply by wchen342** [reply]
I will see whether I can come up with a patch to put the connectivity hint behind a flag. Shouldn't be too hard (I hope).

**Reply by uazo** [reply]
> I will see whether I can come up with a patch to put the connectivity hint behind a flag. 

in my opinion, it's not worth it, unfortunately.
other programs also rely on that connectivity control mechanism: for example I also use DavX5 which refuses to sync if android says there is no connectivity.
if someone uses the device for anything other than the browser, he must necessarily grant access to that android service.

**Reply by csagan5** [reply]
The logic is here: https://source.chromium.org/chromium/chromium/src/+/main:chrome/android/java/src/org/chromium/chrome/browser/net/connectivitydetector/ConnectivityDetector.java;l=144;drc=917a2b5c9c82f96f685692202fd44725b02f6e98

You can see there also the captive portal checks that I mentioned. It uses Android's connectivity manager on Android M+, there used to be a flag to force HTTP checks: https://github.com/chromium/chromium/commit/7d69cf98bd90e83b5f2472b413def908c7ab2dcc

It will fall back to sending HTTP probes on Android < M, but in Bromite I am going to disable that possibility by making `shouldSkipHttpProbes()` always send true (this will also assume that connection is validated on Android < M).

If you want to disable the dialog altogether then the approach should disable parts of the `OfflineDetector` class, but I think there is a much simpler approach: see `OfflineIndicator` and `OfflineIndicatorV2` features.

**Reply by wchen342** [reply]
@b1k3rdude The `OfflineIndicator` is now disabled in v96.

**Reply by b1k3rdude** [reply]
> @b1k3rdude The `OfflineIndicator` is now disabled in v96.

Thank you, this is very much appreciated :-)
